cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -341,13 +354,20 @@ impl<State> CoreEnvironment<State> {
             out_link_path.as_ref().display()
         );
 
+        let service_config_path = if let LockedManifest::Catalog(ref lockfile) = lockfile {
+            maybe_make_service_config_file(flox, lockfile)?
+        } else {
+            None
+        };
+
```

Comment:
nonblocking greenkeeping: related to duplicating logic in `build` and `link`, I don't think it makes sense for `build` to depend on something outside of the lockfile but then be defined on `LockedManifest`. https://github.com/flox/flox/issues/1706 is also related - build should probably use the lockfile itself, not just the contents, which is accessible in `CoreEnvironment` but not `LockedManifest`

Reply:
I'm not sure what "depend on something" and "then be defined" are referring to here. Could you explain more?

Reply:
From sync: look into moving `LockedManifest::build` and it's `pkgdb` calls up into `CoreEnvironment`.